### DISCUSSION TITLE
Wazuh Service fails to start on sensor node, port 55000 allocated to docker-proxy

### DISCUSSION THREAD

**Discussion by ghost** [Security-Onion-Solutions/securityonion#4570] [Unsupported Versions]
Hello All,
After routine update nodes via yum and soup (2.3.52) , then rebooted nodes (Manager, Search, Sensor). 
Sensor node has stopped sending data, listed as fault in Grid, no abnormal issues seen in Graphana.

Did a salt "*" state.highstate and one job returns as failed.

Seems that Wazuh-Agent service cannot restart since port 55000 is already being used by process docker-proxy.

here is the output from the state.hoghstate (redacted) ...
----------
          ID: wazuhagentservice
    Function: service.running
        Name: wazuh-agent
      Result: False
     Comment: Running scope as unit run-21538.scope.
              Job for wazuh-agent.service failed because the control process exited with error code. See "systemctl status wazuh-agent.service" and "journalctl -xe" for details.
     Started: 20:26:19.853416
    Duration: 10363.867 ms
     Changes:
-------
          ID: wazuhportavailable
    Function: cmd.run
        Name: netstat -utanp | grep ":55000" | grep -qv docker && PROCESS=$(netstat -utanp | grep ":55000" | uniq) && echo "Another process ($PROCESS) appears to be using port 55000.  Please terminate this process, or reboot to ensure a clean state so that the Wazuh API can start properly." && exit 1 || exit 0
      Result: True
     Comment: Command "netstat -utanp | grep ":55000" | grep -qv docker && PROCESS=$(netstat -utanp | grep ":55000" | uniq) && echo "Another process ($PROCESS) appears to be using port 55000.  Please terminate this process, or reboot to ensure a clean state so that the Wazuh API can start properly." && exit 1 || exit 0" run
     Started: 20:26:17.734770
    Duration: 223.647 ms
     Changes:
              ----------
              pid:
                  21316
              retcode:
                  0
              stderr:
              stdout:

--------
sudo netstat -utanp | grep ":55000"
tcp        0      0 0.0.0.0:55000           0.0.0.0:*               LISTEN      21339/docker-proxy
-------
Already rebooted the Sensor node, and restarted Wazuh so-wazuh restart without success.

Any suggestions?

Thanks,
Edgar

**Comment by weslambert** [comment]
We have a process in place to reserve ports so this situation does not occur, but it sounds like it is not working as intended.

What is the output of the following from the node?

`sysctl net.ipv4.ip_local_reserved_ports`

`cat /etc/sysctl.d/99-reserved-ports.conf`

**Reply by ghost** [reply]
Thank you for your answer Wes, 
Here is the output of the commands from the Sensor Node ...

$ sudo sysctl net.ipv4.ip_local_reserved_ports
net.ipv4.ip_local_reserved_ports = 47760-47860,55000,57314

$ cat /etc/sysctl.d/99-reserved-ports.conf
net.ipv4.ip_local_reserved_ports=55000,57314,47760-47860

Thanks,
Edgar
